src/harness/testcases/WINNF_FT_S_SIQ_testcase.py

```diff
@@ -841,6 +842,237 @@ def test_WINNF_FT_S_SIQ_11(self):
+        # Inject PPA into SAS UUT
+        zone_id = self._sas_admin.InjectZoneData({"record": ppa['ppaRecord']})
+    
+ 
+    # Step5: Trigger CPAS activity
+    if ((('gwpzRecordsN1' in config) and (config['gwpzRecordsN1'])) or \
+        (('palRecordsN3' in config) and (config['palRecordsN3']))):
+      self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete() 
+
+    # Step6: Send N2 spectrum inquiry requests (one per registered CBSD)
+    spectrum_inquiry_list = config['spectrumInquiryRequestsN2']
+    addCbsdIdsToRequests(cbsd_ids, spectrum_inquiry_list)
+ 
+    spectrum_inquiry_request = {'spectrumInquiryRequest': spectrum_inquiry_list}
+    spectrum_inquiry_response = self._sas.SpectrumInquiry(spectrum_inquiry_request)['spectrumInquiryResponse']
+
+    # Check the response contains N2 SpectrumInquiryResponse objects
+    self.assertEqual(len(spectrum_inquiry_response),len(config['spectrumInquiryRequestsN2']))
+
+    for index,response in enumerate(spectrum_inquiry_response):
+      # If the corresponding request contained a valid cbsdId, the
+      # response shall contain the same cbsdId.
+      if 'cbsdId' in spectrum_inquiry_list[index]:
+        if spectrum_inquiry_list[index]['cbsdId'] in cbsd_ids:
+          self.assertEqual(response['cbsdId'], spectrum_inquiry_list[index]['cbsdId'])
+        else:
+          self.assertFalse('cbsdId' in response)
+
+      # Check response code matches the expected response code
+      self.assertIn(response['response']['responseCode'],config['expectedResponseCodes'][index])
+
+      if (response['response']['responseCode'] == 0) :
+        # Check if the 'availableChannel' parameter is present
+        self.assertTrue('availableChannel' in response)
+
+        # Check if the ruleApplied field is set to FCC_PART_96
+        for channel in response['availableChannel']:
+          self.assertEqual(channel['ruleApplied'], 'FCC_PART_96')
+
+        # Check that the GAA channel frequency ranges do not overlap with FR1cbsd ranges
+        for channel in response['availableChannel']:
+          if channel['channelType'] == 'GAA':
+            for fr1_range in config['fr1Cbsd'][index]:
+              # Check if the channel range is lower than the low freq of FR1cbsd range or if it is higher 
+              # than the high freq. If not then it is considered to be overlapping
+              self.assertTrue(((channel['frequencyRange']['highFrequency'] <= fr1_range['frequency']['lowFrequency']) or 
+                               (channel['frequencyRange']['lowFrequency'] >= fr1_range['frequency']['highFrequency'])), 
+                                "GAA channel overlaps with FR1cbsd")
+
+        # Check the PAL channels frequency ranges overlap with FR2cbsd
+        for channel in response['availableChannel']:
+          if channel['channelType'] == 'PAL':
+            isFrequencyIncludedInFr2Cbsd = False
+            for fr2_range in config['fr2Cbsd'][index]:
```

Comment:
I think you have to use something like `assertChannelsContainFrequencyRange`. I think it is valid for the SAS to respond with e.g. two 5 MHz channels instead of (or in addition to) one 10 MHz channel.

Reply:
Used assertChannelsContainFrequencyRange. May need some clarification if the changes are not as expected. Thanks!